assets/js/graphql/get-traits.js

```diff
@@ -0,0 +1,160 @@
+import { query } from './query.js';
+//import { modalLink } from './modal.js';
+
+
+/** The GraphQL query used to get traits. */
+export const getTraitsQuery = `
+query TraitQuery($pageSize: Int, $page: Int, $name: String, $studyType: String, $publicationId: String, $author: String) {
+    traits(pageSize: $pageSize, page: $page, name: $name, studyType: $studyType, publicationId: $publicationId, author: $author) {
+      results {
+        name
+        qtlStudy {
+          identifier
+          synopsis
+          description
+          genotypes
+          organism {
+            genus
+            species
+          }
+          dataSet {
+            publication {
+              firstAuthor
+              pubMedId
+              doi
+            }
+          }
+        }
+        gwas {
+          identifier
+          synopsis
+          description
+          genotypes
+          organism {
+            genus
+            species
+          }
+          dataSet {
+            publication {
+              firstAuthor
+              pubMedId
+              doi
+            }
+          }
+        }
+      }
+      pageInfo {
+        currentPage
+        pageSize
+        numResults
+        pageCount
+        hasPreviousPage
+        hasNextPage
+      }
+    }
+  }
+`;
+
+
+/**
+ * Gets traits from GraphQL.
+ * @param {object} queryData - An object containing zero or more variables for the GraphQL query.
+ * @param {object} pageData - An object containing pagination data for the GraphQL query, if any.
+ * @param {object} options - An object containing optional parameters for the HTTP request,
+ * namely, an optional `AbortSignal` instance that can be used to cancel the request mid-flight.
+ * @returns {Promise} A `Promise` that resolves to the result of the GraphQL query.
+ */
+export function getTraits(queryData={}, pageData={}, options={}) {
+  const genus = queryData.genus;
```

Comment:
Use [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) to avoid attribute `not defined` errors here.